src/radical/pilot/umgr/staging_input/default.py

```diff
@@ -338,12 +340,23 @@ def _handle_unit(self, unit, actionables):
             src = complete_url(src, src_context, self._log)
             tgt = complete_url(tgt, tgt_context, self._log)
 
-            if rpc.CREATE_PARENTS in flags:
-                copy_flags = rs.filesystem.CREATE_PARENTS
-            else:
-                copy_flags = 0
+            # Check if the src is a folder, if true
+            # add recursive flag if not already specified
+            if os.path.isdir(src.path):
+                if rs.filesystem.RECURSIVE not in flags:
+                    flags += [rs.filesystem.RECURSIVE]
```

Comment:
Why do we need the `RECURSIVE` flag if its always added as needed anyway?

Reply:
If you are commenting that we don't need the recursive flag at the user API, I agree.

Reply:
Ok, assuming that user does not have specify CREATE_PARENTS and RECURSIVE flags. Even if the user does specify, it will not break the staging (64 | 64 = 64).